play/type-racer/app.js

```diff
@@ -66,6 +85,55 @@ function adjustPlayerAndRoad() {
     road.style.height = 150 + "px";
 }
 
+/**
+ * Read in the wordlist text file in order to use its contents as words in the game
+ */
+async function readTextFile(file) {
+
+    let response = await fetch(file);
+
+    // if HTTP-status is valid
+    if (response.ok) {
+
+        // get the response body
+        let textInput = await response.text();
+        words = [];
+        let tempWordIndex = 0;
+        let currentWord = "";
+
+        // loop through the entire string retrieved from the text file
+        for (let i = 0; i < textInput.length; i++) {
```

Comment:
This is a promise-based example (instead of async...await as you've been doing), but this code could be simplified by splitting the file text by newlines.
```js
fetch(WORD_LIST)
    .then(response => response.text())
    .then(data => {
          let words = data.split("\n");
    });
```